addons/stock/models/stock_inventory.py

```diff
@@ -153,6 +158,22 @@ def action_reset_product_qty(self):
         self.mapped('line_ids').write({'product_qty': 0})
         return True
 
+    def action_validate(self):
+        track_line_ids = self.line_ids.filtered(lambda x: x.product_id.tracking in ['lot','serial'] and not x.prod_lot_id)
```

Comment:
- lambda l: l.product_...
- space after the coma
- problematic_tracked_line_ids? not sure of a good variable name.. so you need to add a comment